### ISSUE TITLE
Orchestrator / Transcoder RPC Timeouts

### ISSUE THREAD

**Issue by j0sh** [livepeer/go-livepeer#576]
**Is your feature request related to a problem? Please describe.**

Currently, orchestrators listen to the same port for gRPC, direct bulk uploads, and LPMS downloads. This has worked well so far with broadcasters. However, this runs into a problem when we attempt to expand the orchestrator's interaction to include transcoders on the same port.

There is a read/write timeout of 8 seconds for HTTP requests. While 8 seconds is a reasonable timeout for broadcasters that are expected to be continuously transmitting and receiving requests, transcoders may be connected to an orchestrator for an extended period of time without being assigned work. This does not work well with gRPC streaming, as the connection would time out after 8 seconds of idleness. 

**Describe the solution you'd like**

There are a few options, in order of preference:

1.  RPC-level keepalive (ping-pong) every few seconds between O/T.
    * Generally a useful thing, but pings every ~8 seconds are a bit too frequent
2. Listen to a separate port
    * Some benefits with separation of concerns, but more "moving parts" wrt network exposure
3. Close and re-establish the connection every 8 seconds
    * Increased traffic (eg, HTTPS handshaking), dropping and re-adding transcoders is slower, harder to establish a baseline for connection reliability
4. Remove the timeout entirely
    * Timeouts are a good thing, especially with uncontrolled clients (eg, broadcasters)

**Comment by yondonfu** [comment]
If we use 2, we would still need to do something about the timeout right? By using a different port than the one used to interact with the broadcaster, we can have a longer timeout value (assuming we want to keep timeouts since they are helpful) that is more appropriate for the interaction between the orchestrator and transcoders, but we would still need to use a keepalive every timeout interval?

**Comment by j0sh** [comment]
Correct, with 2, we can set the timeout to whatever's appropriate apart from the broadcaster, or disable the O/T timeout entirely. Running a new port increases the operational surface area though, and it affects how folks interact with the node. I can see benefits either way though, eg internal transcoders talking to a private interface, and broadcasters talking to the public one.

**Comment by ericxtang** [comment]
Option 2 feels reasonable.  B/O interaction should have different timeout than O/T interaction.
